io/plugins/eu.esdihumboldt.hale.io.shp.test/src/eu/esdihumboldt/hale/io/shp/ShapefileInstanceWriterTest.groovy

```diff
@@ -1432,8 +1463,9 @@ class ShapefileInstanceWriterTest {
 
 					assert jts instanceof MultiPolygon
 					def name = inst.p.name.value()
-					assert name
-					switch (name) {
+					String dencodedString = new String(name.getBytes(shapeFileDataStore.getCharset()));
+					assert dencodedString
+					switch (dencodedString) {
```

Comment:
It would be great if here as well in the default case you can mention `decodedString` instead of `typeName`